### DISCUSSION TITLE
Why does all output from a container go to stdout when using a container TTY?

### DISCUSSION THREAD

**Discussion by LewisGaul** [containers/podman#22563] [Q&A]
This works as expected:
```
   > podman run --rm ubi8 bash -c 'echo hello; foo' 1>1.txt 2>2.txt
   > cat 1.txt
hello
   > cat 2.txt
bash: foo: command not found
```

But why when there's a container TTY involved does it all go to stdout in the terminal that runs podman?
```
   > podman run --rm -t ubi8 bash -c 'echo hello; foo' 1>1.txt 2>2.txt
   > cat 1.txt
hello
bash: foo: command not found
   > cat 2.txt
```

**Comment by arizvisa** [comment]
It's as specified per the documentation: https://github.com/containers/podman/blob/main/docs/source/markdown/options/tty.md. The terminal option is specifically for interacting with the container'd process where you need to guarantee the pipes for stdout/stderr are synchronized and ordered correctly.

If you were looking for the code to verify why, I think it might be at https://github.com/containers/podman/blob/main/libpod/oci_conmon_exec_common.go or https://github.com/containers/podman/blob/main/libpod/oci_conmon_common.go. There were some issues on tty and interactivity from a \_really\_ long time ago if you're curious about the history (which I wandered into when contributing a PR).

**Reply by LewisGaul** [reply]
Thanks for the response. I realised this behaviour is the same if running the same over ssh forcing a TTY. Do you have any more context on why this behaviour would be necessary and/or desirable, what are the possible issues with stdout/stderr synchronisation?
